src/proxy-page/steps/addNewSlides.ts

```diff
@@ -0,0 +1,182 @@
+import { ConfigService } from '@nestjs/config';
+import * as cheerio from 'cheerio';
+import { Content } from '../../confluence/confluence.interface';
+import { ContextService } from '../../context/context.service';
+import { Step } from '../proxy-page.step';
+import {
+  getAttribiutesFromChildren, getMacroSlideSettingsPropertyValueByKey, getObjectFromStorageXMLForPageProperties, loadStorageContentToXML,
+} from '../utils/macroSlide';
+
+export default (config: ConfigService, content: Content): Step => (context: ContextService): void => {
+  context.setPerfMark('addNewSlides');
+
+  const commonUnexpectedExpression = [
+    '[data-macro-name="panel"]',
+    '[data-macro-name="info"]',
+    '[data-macro-name="tip"]',
+    '[data-macro-name="note"]',
+    '[data-macro-name="warning"]',
+  ];
+
+  const unexpectedExpressionForParagraphs = [
+    'li',
+    'ul',
+    'ol',
+    'blockquote',
+    ...commonUnexpectedExpression,
+  ];
+
+  const unexpectedExpressionForSpan = [
+    ...commonUnexpectedExpression,
+  ];
+
+  const webBasePath = config.get('web.absoluteBasePath');
+
+  const attachmentResource = `${webBasePath}/wiki/download/attachments/${context.getPageId()}/`;
+
+  const $ = context.getCheerioBody();
+
+  const storageContentXML = loadStorageContentToXML(content);
+
+  const macroSettingsSlideTransition = getMacroSlideSettingsPropertyValueByKey(storageContentXML, 'slide_settings_transition', 'slide');
+
+  const convertSlideFragmentValueToBoolean = (value: string) => value === 'yes';
+
+  const callbackToAssignFragmentClass = (element: cheerio.Element, possibleNested: boolean, forceAssign: boolean) => {
+    if (forceAssign) {
+      $(element).addClass('fragment');
+    }
+    const existRealChildrenWithText = element.children.some((value: cheerio.Node & { data: string; children: { data: string }[] }) => {
+      if (possibleNested) {
+        return value.data && value.data.trim().length > 0;
+      }
+      return true;
+    });
+    if (existRealChildrenWithText) {
+      const htmlElemenet = $(element);
+      const finalElement = (htmlElemenet && htmlElemenet['0']) ?? htmlElemenet;
+      $(finalElement).addClass('fragment');
+    }
+  };
+
+  const searchByTagToAssignFragment = (expression: string, possibleNested: boolean, slideProperties: cheerio.Element, unexpectedExpressionForNestedElements: string[], forceAssign: boolean) => {
+    $(slideProperties).find(expression).each((_: number, element: cheerio.Element) => {
+      if (possibleNested) {
+        const isCorrectConditionalToAssignFragment = unexpectedExpressionForNestedElements.some((expression) => $(element).parents(expression)?.length > 0);
```

Comment:
This line has a length of 156. Maximum allowed is 140.eslint[max-len](https://eslint.org/docs/rules/max-len)

'expression' is already declared in the upper scope on line 62 column 40.eslint[@typescript-eslint/no-shadow](https://typescript-eslint.io/rules/no-shadow)
(parameter) expression: string